pkg/webui/console/containers/gateway-overview-header/index.js

```diff
@@ -0,0 +1,170 @@
+import sharedMessages from '@ttn-lw/lib/shared-messages'
+import attachPromise from '@ttn-lw/lib/store/actions/attach-promise'
+import { selectFetchingEntry } from '@ttn-lw/lib/store/selectors/fetching'
+import { composeDataUri, downloadDataUriAsFile } from '@ttn-lw/lib/data-uri'
+
+import {
+  ADD_BOOKMARK_BASE,
+  addBookmark,
+  DELETE_BOOKMARK_BASE,
+  deleteBookmark,
+} from '@console/store/actions/user-preferences'
+
+import { selectUser } from '@console/store/selectors/logout'
+import { selectBookmarksList } from '@console/store/selectors/user-preferences'
+
+import style from './gateway-overview-header.styl'
+
+const m = defineMessages({
+  addBookmarkFail: 'There was an error and the gateway could not be bookmarked',
+  duplicateGateway: 'Duplicate gateway',
+  removeBookmarkFail: 'There was an error and the gateway could not be removed from bookmarks',
+})
+
+const GatewayOverviewHeader = ({ gateway }) => {
+  const dispatch = useDispatch()
+  const { ids, name, created_at } = gateway
+  const { gateway_id } = ids
+  const user = useSelector(selectUser)
+  const bookmarks = useSelector(selectBookmarksList)
+  const addBookmarkLoading = useSelector(state => selectFetchingEntry(state, ADD_BOOKMARK_BASE))
+  const deleteBookmarkLoading = useSelector(state =>
+    selectFetchingEntry(state, DELETE_BOOKMARK_BASE),
+  )
+
+  const isBookmarked = useMemo(
+    () => bookmarks.map(b => b.entity_ids?.gateway_ids?.gateway_id).some(b => b === gateway_id),
+    [bookmarks, gateway_id],
+  )
+
+  const handleAddToBookmark = useCallback(async () => {
+    try {
+      if (!isBookmarked) {
+        await dispatch(attachPromise(addBookmark(user.ids.user_id, { gateway_ids: ids })))
+        return
+      }
+      await dispatch(
+        attachPromise(
+          deleteBookmark(user.ids.user_id, {
+            name: 'gateway',
+            id: gateway_id,
+          }),
+        ),
+      )
+    } catch (e) {
+      toast({
+        title: gateway_id,
+        message: isBookmarked ? m.removeBookmarkFail : m.addBookmarkFail,
+        type: toast.types.ERROR,
+      })
+    }
+  }, [dispatch, gateway_id, ids, isBookmarked, user.ids])
+
+  const handleGlobalConfDownload = useCallback(async () => {
+    try {
+      const globalConf = await tts.Gateways.getGlobalConf(gateway_id)
+      const globalConfDataUri = composeDataUri(JSON.stringify(globalConf, undefined, 2))
+      downloadDataUriAsFile(globalConfDataUri, 'global_conf.json')
+    } catch (err) {
+      toast({
+        title: sharedMessages.globalConfFailed,
+        message: sharedMessages.globalConfFailedMessage,
+        type: toast.types.ERROR,
+      })
+    }
+  }, [gateway_id])
+
+  const menuDropdownItems = (
+    <>
+      <Dropdown.Item title={sharedMessages.downloadGlobalConf} action={handleGlobalConfDownload} />
+      <Dropdown.Item title={m.duplicateGateway} action={() => {}} />
+      <Dropdown.Item title={sharedMessages.deleteGateway} action={() => {}} />
```

Comment:
You should be able to connect that already. After deletion the user should be sent back to the gateway list with a toast displaying.